collector/netframework_clrloading.go

```diff
@@ -83,114 +89,134 @@ func NewNETFramework_NETCLRLoadingCollector() (Collector, error) {
 			[]string{"process"},
 			nil,
 		),
+		processWhitelistPattern: commonFlags.whitelistRegexp,
+		processBlacklistPattern: commonFlags.blacklistRegexp,
 	}, nil
 }
 
 // Collect sends the metric values for each metric
 // to the provided prometheus Metric channel.
-func (c *NETFramework_NETCLRLoadingCollector) Collect(ctx *ScrapeContext, ch chan<- prometheus.Metric) error {
-	if desc, err := c.collect(ch); err != nil {
-		log.Error("failed collecting win32_perfrawdata_netframework_netclrloading metrics:", desc, err)
+func (c *NETFrameworkCLRLoadingCollector) Collect(ctx *ScrapeContext, ch chan<- prometheus.Metric) error {
+	if desc, err := c.collect(ctx, ch); err != nil {
+		log.Error("failed collecting netframework_clrloading metrics:", desc, err)
 		return err
 	}
 	return nil
 }
 
-type Win32_PerfRawData_NETFramework_NETCLRLoading struct {
+type netframeworkCLRLoading struct {
 	Name string
 
-	AssemblySearchLength      uint32
-	BytesinLoaderHeap         uint64
-	Currentappdomains         uint32
-	CurrentAssemblies         uint32
-	CurrentClassesLoaded      uint32
-	PercentTimeLoading        uint64
-	Rateofappdomains          uint32
-	Rateofappdomainsunloaded  uint32
-	RateofAssemblies          uint32
-	RateofClassesLoaded       uint32
-	RateofLoadFailures        uint32
-	TotalAppdomains           uint32
-	Totalappdomainsunloaded   uint32
-	TotalAssemblies           uint32
-	TotalClassesLoaded        uint32
-	TotalNumberofLoadFailures uint32
+	AssemblySearchLength      float64 `perflib:"Assembly Search Length"`
+	BytesinLoaderHeap         float64 `perflib:"Bytes in Loader Heap"`
+	Currentappdomains         float64 `perflib:"Current appdomains"`
+	CurrentAssemblies         float64 `perflib:"Current Assemblies"`
+	CurrentClassesLoaded      float64 `perflib:"Current Classes Loaded"`
+	PercentTimeLoading        float64 `perflib:"% Time Loading"`
+	Rateofappdomains          float64 `perflib:"Rate of appdomains"`
+	Rateofappdomainsunloaded  float64 `perflib:"Rate of appdomains unloaded"`
+	RateofAssemblies          float64 `perflib:"Rate of Assemblies"`
+	RateofClassesLoaded       float64 `perflib:"Rate of Classes Loaded"`
+	RateofLoadFailures        float64 `perflib:"Rate of Load Failures"`
+	TotalAppdomains           float64 `perflib:"Total Appdomains"`
+	Totalappdomainsunloaded   float64 `perflib:"Total appdomains unloaded"`
```

Comment:
Is the mixed cases of `Appdomains`/`appdomains` correct?

Reply:
Yes, the mixed casing is correct. We could change the name of the struct members, of course, but those are exactly how the performance counters are spelled in Windows.